Makefile

```diff
@@ -122,8 +119,15 @@ MMCTL_REL_TO_DOWNLOAD = $(shell scripts/get_latest_release.sh 'mattermost/mmctl'
 
 all: run ## Alias for 'run'.
 
+-include config.override.mk
```

Comment:
I'm still confused by `config.override.mk` and `config.mk`:
* We import the /override/ before we import the canonical `config.mk`, relying on the conditional assignment (`?=`) to avoid undoing any overrides. This is brittle to a future change to `config.mk` that doesn't realize the nuances here. 
* As noted before, I can't make decisions from my override script using the canonical definitions of these variables since they aren't defined yet.
* I don't understand the subset of things that were moved to `config.mk`, nor can I explain to someone under what conditions a future variable should reside in the `Makefile` or in `custom.mk`.

I remain inclined to think we should eliminate `config.mk` (moving the variables back here), import `build/*.mk` and finally import a `override.mk` (or more canonically, a `build/custom.mk` like our plugin repositories).

Thoughts?

Reply:
moving that into `build/custom.mk` make it hard to find for any new developer. Having a `config.mk` shows explicitly that you have certian things that we consider configuration of your development environment, and then `config.override.mk` make it pretty clear what is your configuration.

About the order, is on purpose, I want you to be able override settings (forcing them or not) and at the same time allowing you to override them with environment variables.

So, If I load the config.mk first, I'm force to define my settings as "mandatory" if I want to override the `config.mk`, and that remove the posibility of having environment variables overwriting that. My idea is, you can "enforce settings" or you can "softly define settings", the solftly defined can be override by environment variables, the enforced ones no.

Reply:
> So, If I load the `config.mk` first, I'm force to define my settings as "mandatory" if I want to override the `config.mk`, and that remove the posibility of having environment variables overwriting that. My idea is, you can "enforce settings" or you can "softly define settings", the solftly defined can be override by environment variables, the enforced ones no.

Got it -- but I admit this is /really/ subtle. Regardless of where we end up, can you update the `README.md` or developer docs to clearly document these decisions?

Are we sure we need to define two semantically different ways of overriding the base configuration? What if we only just supported environment variable overrides, with everything in `Makefile` being a `?=`? Less discoverable, but maybe less surprising.

Are there open source repositories using this strategy that might help clarify my understanding of the vision here? I'm worried this feels too "cobbled together" to be unambiguously useful.

Reply:
The environment variables is hard to discover, really hard, and is not so easy to manage, having a config file that you have in your filesystem is way easier and comfortable. I don't see why it should be surprising? And having this files, instead of a set of variables in the Makefile is more clear an self-documented than "use environment variables to override things there in the Makefile and see what happens".

Reply:
I remain surprised by the proposed behaviour, but I'm happy to withdraw my requested changes. There's lots of great stuff here that I hope to use soon :)